lib/Models/ArcGisPortalItemReference.ts

```diff
@@ -6,6 +6,7 @@ import URI from "urijs";
 import isDefined from "../Core/isDefined";
 import { JsonObject } from "../Core/Json";
 import loadJson from "../Core/loadJson";
+import AccessControlMixin from "../ModelMixins/AccessControlMixin";
```

Comment:
What does the AccessControlMixin do here in the Reference version of the CatalogGroup?

Reply:
It's so that we can call the `setAccessType` function in `ArcGisPortalCatalogGroup`, which allows us to give the catalog item an `access` property (`"public"/"private"/etc`). Basically with this mixin for each catalog item we can assign it an access value based on its access control settings in portal. By default `AccessControlMixin` is already incorporated in a `CatalogMemberMixin`, so I didn't have to add it to `ArcGisPortalCatalogGroup`, but not in references like `MagdaReference` and `ArcGisPortalItemReference`, so I had to add that mixin (just like how it was done in `MagdaReference`).